csharp/ql/src/semmle/code/csharp/Preprocessor.qll

```diff
@@ -236,3 +236,95 @@ class EndRegionDirective extends PreprocessorDirective, @directive_endregion {
 
   override string getAPrimaryQlClass() { result = "EndRegionDirective" }
 }
+
+/**
+ * A branching preprocessor directive, such as `IfDirective`, `ElifDirective`, or
+ * `ElseDirective`.
+ */
+class BranchDirective extends PreprocessorDirective {
```

Comment:
This is missing a base class, e.g. `BranchDirective_` where `private class BranchDirective_ = @directive_if or @directive_elif or @directive_else`.